modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevokeWithInvalidClientCredentialsTestCase.java

```diff
@@ -57,26 +62,63 @@ public void testInit() throws Exception {
-     * After cache timeout new token should issued after it revoked
+     * This test is written to test for token revocation with invalid client id , secrets and access tokens.
      *
      * @throws Exception
      */
     @SetEnvironment(executionEnvironments = { ExecutionEnvironment.ALL })
-    @Test(description = "Revoke token with invalid sending of client credentials.")
-    public void testRevokeTokenAfterCacheTimedOut() throws Exception {
-        String clientKey = "dummyConsumerKey";
-        String clientSecret = "dummyConsumerSecret";
-        String accessToken = "dummyAccessToken";
+    @Test(description = "Revoke token with invalid credentials.")
+    public void testRevokeToken() throws Exception {
+        //Application utils
+        OAuthConsumerAppDTO appDto = createApplication();
+        consumerKey = appDto.getOauthConsumerKey();
+        consumerSecret = appDto.getOauthConsumerSecret();
+        String errorMessage = null;
+        //request for token
+        String token = requestAccessToken(consumerKey, consumerSecret, TOKEN_API_ENDPOINT);
+
+        //Revoke access token with invalid format of client credentials
+        errorMessage = revokeToken(token, getInvalidBase64EncodedString(consumerKey, consumerSecret));
+        Assert.assertEquals(OAuth2Constant.INVALID_CLIENT, errorMessage,
+                "Invalid format in sending client credentials, should have produced : " + OAuth2Constant.INVALID_CLIENT
+                        + "error code");
+
+        //client credentials been null
+        errorMessage = revokeToken(token, null);
+        Assert.assertEquals(OAuth2Constant.INVALID_CLIENT, errorMessage,
+                "Invalid format in sending client credentials, should have produced : " + OAuth2Constant.INVALID_CLIENT
+                        + "error code");
+
+        //revoke with invalid client key
+        errorMessage = revokeToken(token, getBase64EncodedString("dummyConsumerKey", consumerSecret));
+        Assert.assertEquals(OAuth2Constant.UNAUTHORIZED_CLIENT, errorMessage,
+                "Invalid consumer key in sending client credentials, should have produced : " + OAuth2Constant.INVALID_CLIENT
+                        + "error code");
+
+        //revoke with invalid client secret
+        errorMessage = revokeToken(token, getBase64EncodedString(consumerKey, "dummyConsumerSecret"));
+        Assert.assertEquals(OAuth2Constant.UNAUTHORIZED_CLIENT, errorMessage,
+                "Invalid consumer secret in sending client credentials, should have produced : " + OAuth2Constant.INVALID_CLIENT
+                        + "error code");
+
+        //token is required parameter in request
+        errorMessage = revokeToken(null, getBase64EncodedString(consumerKey, consumerSecret));
+        //The request is missing a required parameter as per spec.
+        Assert.assertEquals(OAuth2Constant.INVALID_REQUEST, errorMessage,
```

Comment:
We can static import of assetEquals() you dont need to call  Assert.assertEquals()